components/experimental/src/dimension/percent/formatter.rs

```diff
@@ -0,0 +1,104 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Experimental.
+
+use fixed_decimal::FixedDecimal;
+use icu_provider::prelude::*;
+
+use super::super::provider::percent::PercentEssentialsV1Marker;
+use super::format::FormattedPercent;
+use super::options::PercentFormatterOptions;
+
+extern crate alloc;
+
+/// A formatter for percent values.
+///
+/// [`PercentFormatter`] supports:
+///   1. Rendering in the locale's percent system.
+pub struct PercentFormatter {
+    /// Essential data for the percent formatter.
+    essential: DataPayload<PercentEssentialsV1Marker>,
+
+    /// Options bag for the percent formatter to determine the behavior of the formatter.
+    options: PercentFormatterOptions,
+}
+
+impl PercentFormatter {
+    icu_provider::gen_any_buffer_data_constructors!(
+        (locale, options: PercentFormatterOptions) -> error: DataError,
+        functions: [
+            try_new: skip,
+            try_new_with_any_provider,
+            try_new_with_buffer_provider,
+            try_new_unstable,
+            Self
+        ]
+    );
+
+    /// Creates a new [`PercentFormatter`] from compiled locale data and an options bag.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub fn try_new(
```

Comment:
Suggestion: an easy way to get this up and running, and something we should support anyway, is passing a FixedDecimalFormatter into the constructor. We do this wrapping in a few other places, where one formatter builds on another. A good one to reference is `PluralRulesWithRanges`.

You can add a type parameter to your struct as so:

```rust
impl<R> PercentFormatter<R>
where
    R: AsRef<FixedDecimalFormatter>,
{
```

And then change the constructor as below:

```rust
    pub fn try_new_with_fixed_decimal_formatter_unstable(
        provider: &(impl DataProvider<PercentEssentialsV1Marker> + ?Sized),
        locale: &DataLocale,
        fixed_decimal_formatter: R,
        options: PercentFormatterOptions,
    ) -> Result<Self, DataError> {
        // ...
    }
```